src/Confluent.SchemaRegistry/Rest/RestService.cs

```diff
@@ -0,0 +1,302 @@
+﻿// Copyright 2016-2018 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System.Collections.Generic;
+using System.Net;
+using System.Linq;
+using System.Net.Http;
+using System;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+
+
+namespace Confluent.SchemaRegistry
+{
+    /// <remarks>
+    ///     It may be useful to expose this publically, but this is not
+    ///     required by the Avro serializers, so we will keep this internal 
+    ///     for now to minimize documentation / risk of API change etc.
+    /// </remarks>
+    internal class RestService : IRestService
+    {
+        private static readonly string acceptHeader = string.Join(", ", Versions.PreferredResponseTypes);
+
+        /// <summary>
+        ///     The index of the last client successfully used (or random if none worked).
+        /// </summary>
+        private int lastClientUsed;
+        private object lastClientUsedLock = new object();
+
+        /// <summary>
+        ///     HttpClient instances corresponding to each provided schema registry Uri.
+        /// </summary>
+        private readonly List<HttpClient> clients;
+
+        
+        /// <summary>
+        ///     Initializes a new instance of the RestService class.
+        /// </summary>
+        public RestService(string schemaRegistryUris, int timeoutMs)
```

Comment:
nit: Uris is probably technically correct, but since url is used in other places it'd be better if they were the same (for grepping, etc)

Reply:
ok, but this whole situation makes me sad.

Reply:
btw. we hardly ever need grep with our fancy C# editors.